assembly/src/parsers/io_ops/mod.rs

```diff
@@ -33,214 +30,46 @@ use mem_ops::*;
 /// Removes the next `n` values from the advice tape and pushes them onto the stack. The number of
 /// items that can be read from the advice tape is limited to 16.
 ///
-/// *ENVIRONMENT: `push.env.{var}`*
-/// Pushes the value of the specified environment variable onto the top of the stack. Currently, the
-/// only environment input is `sdepth`.
-///
-/// *RANDOM ACCESS MEMORY: `push.mem`, `push.mem.a`*
-/// Pushes the first element of the word at memory address `a` onto the stack. If no memory address
-/// is specified, it is assumed to be on top of the stack.
-///
-/// *LOCAL PROCEDURE VARIABLES: `push.local.i`*
-/// Pushes the first element of the word at the local memory address with index `i` onto the stack.
-///
 /// # Errors
 ///
 /// Returns an `AssemblyError` if the op param is invalid, malformed, or doesn't match an expected
 /// `push` instruction
-pub fn parse_push(
-    span_ops: &mut Vec<Operation>,
-    op: &Token,
-    num_proc_locals: u32,
-) -> Result<(), AssemblyError> {
+pub fn parse_push(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
     if op.num_parts() < 2 {
         return Err(AssemblyError::invalid_op(op));
     }
     if op.parts()[0] != "push" {
         return Err(AssemblyError::unexpected_token(
             op,
-            "push.{adv.n|env.var|local.i|mem|mem.a|a|a.b|a.b.c...}",
+            "push.{adv.n|a|a.b|a.b.c...}",
         ));
     }
```

Comment:
Let's convert this to `debug_assert_eq` similar to how I mentioned in the other comments. I'd also move this up to be before `num_parts` check.